### DISCUSSION TITLE
REST API not working using curl from command prompt - Linux

### DISCUSSION THREAD

**Discussion by RicardoFernandez-UY** [Apicurio/apicurio-registry#2287] [Q&A] [answered]
Hi,
I'm just starting to use apicurio-registry and I Installed it on Docker following the instructions here:

(https://www.apicur.io/registry/docs/apicurio-registry/2.1.x/getting-started/assembly-installing-registry-docker.html#installing-registry-in-memory-storage_registry)

The registry is working because the UI shows up and I've been able to load some definitions from there.
But when I try to work with the API by sending curl commands from the command prompt:
If I issue a GET request it always returns an empty body
And if I issue a POST request it says that "HTTP method POST is not supported by this URL" for example:

```
curl -X POST -H "Content-type: application/json; artifactType=AVRO" \
  -H "X-Registry-ArtifactId: share-price" \
  --data '{"type":"record","name":"price","namespace":"com.example", \
   "fields":[{"name":"symbol","type":"string"},{"name":"price","type":"string"}]}' \
  http://localhost:8081/apis/registry/v2/groups/my-group/artifacts
```

Will return:

<html><head><title>Error</title></head><body>HTTP method POST is not supported by this URL</body></html>

I guess I might be doing something wrong. 
I will really appreciate any help on this issue.

Thank you!!!

**Comment by EricWittmann** [comment, marked as answer]
Can you include the docker command you used to start registry?  I just tested this locally and your `curl` command worked perfectly for me.  Here's what I did:

`docker run -it -p 8081:8080 apicurio/apicurio-registry-mem:latest-release`

And then I just copy/pasted your command (and piped it to `jq`):

```
$ curl -X POST -H "Content-type: application/json; artifactType=AVRO"   -H "X-Registry-ArtifactId: share-price"   --data '{"type":"record","name":"price","namespace":"com.example", \
   "fields":[{"name":"symbol","type":"string"},{"name":"price","type":"string"}]}'   http://localhost:8081/apis/registry/v2/groups/my-group/artifacts | jq
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100   366  100   224  100   142   2776   1759 --:--:-- --:--:-- --:--:--  4575
{
  "createdBy": "",
  "createdOn": "2022-03-02T12:59:47+0000",
  "modifiedBy": "",
  "modifiedOn": "2022-03-02T12:59:47+0000",
  "id": "share-price",
  "version": "1",
  "type": "AVRO",
  "globalId": 2,
  "state": "ENABLED",
  "groupId": "my-group",
  "contentId": 2
}
```

Works!

Try this:

`curl http://localhost:8081/apis/registry/v2/system/info`

Does that work?

**Reply by RicardoFernandez-UY** [reply]
Hi Eric,
Thank you very much for your answer!!

I tried it; it didn't work.
Then I noticed the only difference in your code was the "latest-release" vs  "latest" which is the one I used to create the container. 
After reading a bit about the fact, I understood that latest will pull the latest no matter if it is stable or not, and latest-release will pull the stable version when creating the container.
In summary, I deleted everything and re-created the container like you did and it worked!! 
So, I think it had something to do with that.

The following is just in case someone hits the same problem.
After solving the issue with the in-memory image, I tried using the Kafka version.
It installed fine but when trying to issue curl commands I started getting other errors (parsing errors when issuing the same curl command we used so far)
I investigated a bit more and saw that the image for Kafka was a year old. Also tried the KafkaSQL and got the same parsing errors.
(things like this:  parse error: Invalid numeric literal at line 1, column 50)

Then I used the SQL-only image and it worked like a charm. This image was updated a day ago.

There might be a need for a little housekeeping on those images as it seems the old ones might be having some issues. At least that was my experience.

I hope this helps!
And thank you again!

**Reply by EricWittmann** [reply]
Thanks for the feedback.  You are right that we probably have some stale docker images in there, especially the `latest` images.  We moved away from `latest` and instead use `latest-release` and `latest-snapshot` in an (clearly failed) attempt to avoid confusion. :)

The kafkasql image should have worked though, as long as you pointed it at a Kafka cluster successfully.  The kafka and streams images are old (we deprecated those and removed them as of registry 2.x).  So now we support three images:

* -mem
* -sql
* -kafkasql

It sounds like the first two worked but the third did not.  If you had the time and could outline exactly what you ran and the errors you got (or even just what you ran) that could be helpful.

Thanks!
